### PR TITLE
Create Blog “getting-started-with-private-cloud-business-edition-apis”

### PR DESCRIPTION
**Pull request by silfhout**
Automatically generated by Netlify CMS

### CODE REVIEW

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+the JSON data holds the access_token itself and also the expiration time, which is 2 hours (7200 seconds).
+
+> *Note*: For simplicity and demo reasons we store the client_id and client_secret in the code, obviously that should never be done in production! How to implement a more secure method will be explained in the next episode of this series.
+
+## Calling an API
+
+A list of available APIs for HPE GreenLake for Private Cloud Business Edition can be found at <https://developer.greenlake.hpe.com/docs/greenlake/services/private-cloud-business/public/> When publishing this blog the APIs were still in beta, however the GA release is planned shortly after.
+
+In this example we will collect an overview of all System Software Catalogs.
```
Comment (choukden) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:103]: we --> you:

In this example you will collect an overview of all _System Software Catalogs_.

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            "updatedAt": "2024-05-03T13:32:55Z",
+            "version": "2.99"
+        }
+    ],
+    "offset": 0,
+    "total": 11
+}
+```
+
+In the next blog in this series we will have a look at a more secure implementation of the client_secret, using a key chain manager to read the client_secret.
```
Comment (choukden) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:365]: Suggestion: 

In the next blog post of this series you will have a look at a more secure...

or

In the next blog post of this series, I will explore a more secure implementation of...

Also, a suggestion to add something like that:

If you have any questions on the HPE GreenLake for Private Cloud Business Edition APIs, or if you are interested sharing your feedback and use cases on this set of APIs, please join the HPE Developer Slack [Workspace](https://developer.hpe.com/slack-signup) and start a discussion in our [#hpe-greenlake-data-services](https://hpedev.slack.com/archives/C02D6H623JP) Slack channel.

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Mark,

Thank you very much for your great contribution. It is really a good blog post for this PCBE series.

I have a few suggestions for you and a tip to assist you with the review in GitHub.

Here are some tips to assist you with the review of the suggestions and how to acknowledge them in GitHub:

- Navigate to **File Changed** area to visualize the comments from start to the end, or navigate to **Conversation** area to visualize the comments per reviewer.

- As you incorporate the suggestions for edit in your blog post, please click on button **Resolve Conversation** in GitHub for each suggestion you incorporated.

**Note** we recommend you to continue doing edits of the blog post entry in the CMS Workflow editor while the entry is in the "**In Review**" column. Moving it back to the "**Drafts**" column will close the Pull Request (PR) in Github and when moving the entry to the “_In Review_” column again a new PR in Github is created, and we will lose the suggestions for edits.

Thanks again!!
Denis

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+They are looking for a cloud-native control plane that scales with infrastructure, so managing hundreds of systems across geographies is as simple as one.
+
+## Simplify infrastructure operations
+
+With cloud agility Hewlett Packard Enterprise provides HPE GreenLake for Private Cloud
+Business Edition delivering VMs across hybrid cloud, on demand. With this offering, customers can build their self-service cloud on demand when they need it and where they need it in data centers, at the edge, and beyond. HPE GreenLake for Private Cloud Business Edition changes the way customers procure and manage VMs with a cloud operational experience. HPE GreenLake for Private Cloud Business Edition helps eliminate complexity with a unified cloud service for VM to infrastructure management, including public cloud VMs.
+The solution — delivered through the Data Services Cloud Console — enables global unified management and monitoring of VMs and infrastructure deployed at any location as well as 1-click, multisite upgrades. This enables IT to reduce operating costs while helping optimize resource utilization, move to a generalist
```
Comment (joe-boyle24) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:34]:
```suggestion
The solution, delivered through Data Services Cloud Console, enables global unified management and monitoring of VMs and infrastructure deployed at any location, It also supports one-click, multisite upgrades. This enables IT to reduce operating costs while helping optimize resource utilization, move to a generalist
```

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+Customers are looking for a radically simplified experience for data infrastructure at scale across the lifecycle — from streamlined device deployment to virtual machine (VM) provisioning, to a global VM and infrastructure management.
+They are looking for a cloud-native control plane that scales with infrastructure, so managing hundreds of systems across geographies is as simple as one.
+
+## Simplify infrastructure operations
+
+With cloud agility Hewlett Packard Enterprise provides HPE GreenLake for Private Cloud
+Business Edition delivering VMs across hybrid cloud, on demand. With this offering, customers can build their self-service cloud on demand when they need it and where they need it in data centers, at the edge, and beyond. HPE GreenLake for Private Cloud Business Edition changes the way customers procure and manage VMs with a cloud operational experience. HPE GreenLake for Private Cloud Business Edition helps eliminate complexity with a unified cloud service for VM to infrastructure management, including public cloud VMs.
```
Comment (joe-boyle24) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:33]:
```suggestion
Business Edition delivering VMs on-demand across hybrid cloud. With this offering, customers can build their self-service cloud on demand when they need it and where they need it in data centers, at the edge, and beyond. HPE GreenLake for Private Cloud Business Edition changes the way customers procure and manage VMs with a cloud operational experience. HPE GreenLake for Private Cloud Business Edition helps eliminate complexity with a unified cloud service for VM to infrastructure management, including public cloud VMs.
```

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+model, and shift from managing infrastructure to managing VMs and data, thereby refocusing resources and skills on higher-value strategic initiatives.
+
+## Automating management
+
+Besides the browser-based Data Services Cloud Console, HPE GreenLake for Private Cloud Business Edition offers a set of Application Programming Interfaces (API's) to automate management of HPE GreenLake for Private Cloud Enterprise or even integrate this with the customer's own management tools. These API's are
+governed by role-based access controls (RBACs) similar to the regular users using the browser-based console. The API's will authenticate using a token, which is created from the client_id and client_secret key pair, and will have the same permissions as the user that created the client_id and client_secret key pair. Each API call will be audited and logged by the HPE GreenLake platform and will be listed under the users id (usually the email address of the user).
```
Comment (joe-boyle24) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:40]:
```suggestion
governed by role-based access controls (RBACs) similar to the regular users using the browser-based console. The API's will authenticate using a token, which is created from the client_id and client_secret key pair, and will have the same permissions as the user that created the client_id and client_secret key pair. Each API call will be audited and logged by HPE GreenLake platform and will be listed under the user's id (usually the email address of the user).
```

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+client = BackendApplicationClient('e21f3028-8097-4a4f-b491-a49b1d102d4a')         
+        
+oauth = OAuth2Session(client=client)         
+auth = HTTPBasicAuth('e21f3028-8097-4a4f-b491-a49b1d102d4a', '05656940014f11ef9946a2408e898685')         
+   
+token = oauth.fetch_token(token_url='https://sso.common.cloud.hpe.com/as/token.oauth2', auth=auth)         
+print(token["access_token"])
+```
+
+> *Note*: Make sure the modules oauthlib, requests and requests_oauthlib are installed in your environment, if not just install these using:
```
Comment (joe-boyle24) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:78]:
```suggestion
> *Note*: Make sure the modules oauthlib, requests and requests_oauthlib are installed in your environment, if not you can install these using:
```

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+}
+```
+
+The JSON data holds the access_token itself and also the expiration time, which is 2 hours (7200 seconds).
+
+> Note: For simplicity and demo reasons we store the client_id and client_secret in the code, obviously that should never be done in production! How to implement a more secure method will be explained in the next episode of this series.
+
+## Calling an API
+
+A list of available APIs for HPE GreenLake for Private Cloud Business Edition can be found at <https://developer.greenlake.hpe.com/docs/greenlake/services/private-cloud-business/public/> When publishing this blog post the APIs were still in beta, however the GA release is planned shortly after.
```
Comment (joe-boyle24) [inline content/blog/getting-started-with-private-cloud-business-edition-apis.md:103]: Is it still the case they're in Beta, or are now GA? Can this be deleted?

**Review by joe-boyle24 — CHANGES_REQUESTED** [review]

File: content/blog/getting-started-with-private-cloud-business-edition-apis.md (hunk 7 above)
Comment (silfhout) [reply content/blog/getting-started-with-private-cloud-business-edition-apis.md:103]: unfortunately they are still in beta... I believe the GA date has been moved to end of May... I will check... If the GA date is before the publishing date I will adjust the text. Else, I will changed the published blog post as soon as the APIs have gone GA...

**Comment by silfhout** [comment]
Do I need to close the pull request in GitHub?

**Pull request merged**